src/App.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
-import './App.scss';
+import { Component } from 'react';
+import { Clock } from './components/Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+import './App.scss';
 
-  return `Clock-${value}`;
+interface State {
+  clockName: string;
+  id: number;
+  hasClock: boolean;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    id: 0,
```

Comment:
if the variable doesn't affect on the content of the component, you can create it directly in the class, not in the state object